programs/exchange_api/src/exchange_processor.rs

```diff
@@ -192,7 +192,7 @@ impl ExchangeProcessor {
         let mut to_account =
             Self::deserialize_account(&keyed_accounts[TO_ACCOUNT_INDEX].account.data)?;
 
-        if &id() == keyed_accounts[FROM_ACCOUNT_INDEX].unsigned_key() {
+        if &faucet_id() == keyed_accounts[FROM_ACCOUNT_INDEX].unsigned_key() {
```

Comment:
Update to `/bench-exchange` to use the new `faucet_id` required?

Reply:
I searched and didn't find anywhere in `/bench_exchange` calling `exchange_instruction::transfer_request` or using `ExchangeInstruction::TransferRequest`, but wouldn't mind your sanity check on that @jackcmay

Reply:
You're right, put an "optimization" in so that token accounts are funded on creation.  When that goes away `faucet_id` will be used.